Fix quotes with syntax properties

MicahChalmer opened the pull request:
This uses syntax properties to make it so that emacs recognizes the single quote, rather than the double quote, as the string delimiter within character literals, while leaving the syntax unchanged elsewhere.

Also included is a commit that restores ERT tests that were inadvertently broken by an earlier commit (removing trailing whitespace broke the tests--I restored them and made it so they are no longer dependent on having trailing whitespace in the code.)

Fixes #1.
Fixes #6.

pnkfelix commented:
Are there any tests of character literal parsing?

I would in particular like to see tests of `'\''` (i.e. the single quote character); (I am inferring that is handled by this code, but I haven't read it quite carefully enough to confirm that.)

MicahChalmer commented:
I added a test for `'\''`.  I also confirmed that this same change also fixes #6, and added a test for that.  (I updated the pull request body to reflect that.)

pnkfelix commented:
@MicahChalmer hey thanks for following up; I took the liberty of cherry-picking the first commit in the series because that seemed important for overall workflow (at least for me).  I'll be looking at the other commits shortly.
